test/integration/src/main/java/io/pravega/test/integration/compatibility/CompatibilityChecker.java

```diff
@@ -221,7 +226,99 @@ private void checkStreamDelete() {
         assertFalse(streamManager.checkStreamExists(scopeName, streamName));
     }
 
-    public static void main(String[] args) throws DeleteScopeFailedException {
+    /**
+     * This method attempts to validate abort functionality after creating a transaction.
+     * @throws TxnFailedException if unable to commit or write to the transaction.
+     */
+    private void checkTransactionAbort() throws TxnFailedException {
+        String scopeName = "transaction-abort-test-scope";
+        String streamName = "transaction-abort-test-stream";
+        streamManager.createScope(scopeName);
+        streamManager.createStream(scopeName, streamName, streamConfig);
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scopeName, ClientConfig.builder().controllerURI(controllerURI).build());
+        @Cleanup
+        TransactionalEventStreamWriter<String> writerTxn = clientFactory.createTransactionalEventWriter(streamName, new UTF8StringSerializer(),
+                EventWriterConfig.builder().build());
+        // Begin a transaction
+        Transaction<String> txn = writerTxn.beginTxn();
+        assertNotNull(txn.getTxnId());
+        // Writing to the transaction
+        txn.writeEvent("event test");
+        // Checking and validating the Transaction status.
+        Transaction.Status status = txn.checkStatus();
+        assertEquals("OPEN", status.toString());
```

Comment:
NIT : Instead of converting status to string, can we do like
`assertEquals(Transaction.Status.OPEN, status.toString());`